crates/cheatcodes/src/inspector.rs

```diff
@@ -1045,14 +1136,22 @@ impl<DB: DatabaseExt> Inspector<DB> for Cheatcodes {
     }
 
     #[inline]
-    fn step_end(&mut self, interpreter: &mut Interpreter, _ecx: &mut EvmContext<DB>) {
+    fn step_end(&mut self, interpreter: &mut Interpreter, ecx: &mut EvmContext<DB>) {
         if self.gas_metering.paused {
             self.meter_gas_end(interpreter);
         }
 
         if self.gas_metering.touched {
             self.meter_gas_check(interpreter);
         }
+
+        // `setArbitraryStorage` and `copyStorage`: add arbitrary values to storage.
+        if (self.arbitrary_storage.is_arbitrary(&interpreter.contract().target_address) ||
+            self.arbitrary_storage.is_copy(&interpreter.contract().target_address)) &&
+            interpreter.current_opcode() == op::SLOAD
```

Comment:
this should be `op::SLOAD` check, please be very mindful of what goes in step and step_end because these functions are called millions of times

Reply:
oh, ok, will do a PR to change this, mind to explain the difference? (was thinking that is better to check if is arbitrary or copy of arbitrary storage first and only after if one of them true if it's a SLOAD, as arbitrary and copy are not so commons)

Reply:
They are called on execution of every single opcode, meaning tens to hundreds of thousands per single execution of a test; we want them inlined and optimized for the common path of "do nothing" as most of these checks will be false most of the time

Reply:
wrt the checks order, the thinking is not correct because checking opcode is 1 or 2 instructions, while the others are hashmap lookups which are in the order of hundreds or thousands

Reply:
thank you, makes total sense, will follow up with a PR to get it inline

Reply:
wonder if better to have arbitrary storage as an option, made a draft PR here https://github.com/foundry-rs/foundry/pull/8848